src/diff/unmount.js

```diff
@@ -42,9 +33,19 @@ export function unmount(internal, parentInternal, skipRemove) {
 
 	if ((r = internal._children)) {
 		for (let i = 0; i < r.length; i++) {
-			if (r[i]) unmount(r[i], parentInternal, skipRemove);
+			if (r[i]) {
+				unmount(
+					r[i],
+					parentInternal,
+					~internal._flags & TYPE_ROOT && internal._flags & TYPE_DOM
```

Comment:
I'm assuming that these bitwise operations are fast enough to do on every child/call to unmount (it saves some bytes!). If they are not, I think it'll show up on 02_replace1k which unmounts and mounts a tree of DOM every iteration.

Reply:
@andrewiggins I think we can pass a constant `1` here, if TYPE_ROOT handling is moved to line 46:

```js
// Line 46:
if (!skipRemove && internal._dom || ~internal._flags & TYPE_ROOT) {
  // either this is skipRemove=0, or a root Internal which we always remove:
  removeNode(internal._dom);
}
```

Reply:
or something like:

```js
	if (internal._flags & TYPE_DOM) {
		if (skipRemove) internal._dom = null;
		else skipRemove = 1;  // only used for children after this point
	}

	if ((r = internal._children)) {
		for (let i = 0; i < r.length; i++) {
			if (r[i]) unmount(r[i], parentInternal, skipRemove);
		}
	}

	// this will only be present for TYPE_DOM that wasn't set to null above due to skipRemove
	if (internal._dom) removeNode(internal._dom);
	internal._dom = null;
```

Reply:
I pushed a corrected version of my suggestions in 69993fd.